packages/eslint-plugin/src/rules/restrict-template-expressions.ts

```diff
@@ -50,7 +50,9 @@ const optionTesters = (
   tester,
 }));
 type Options = [
-  { [Type in (typeof optionTesters)[number]['option']]?: boolean },
+  { [Type in (typeof optionTesters)[number]['option']]?: boolean } & {
+    ignoredTypeNames?: string[];
```

Comment:
🤔 the other options are named like `allow*`. Other places that use the standard format:

* https://typescript-eslint.io/rules/prefer-readonly-parameter-types#allow: `allow`
* https://github.com/typescript-eslint/typescript-eslint/pull/9186: `allowForKnownSafePromises`

Neither of those options mention "type" or "name", because it's redundant in the options' contexts. I'd vote for this rule's option being just `allow`.

Reply:
I'm fine with this. However, I will point out that the option in [`no-base-to-string`](https://typescript-eslint.io/rules/no-base-to-string) is called `ignoredTypeNames`, and `no-base-to-string` is a much more similar rule to this one than the two rules you've mentioned above.